app/controllers/concerns/reauthentication_required_concern.rb

```diff
@@ -7,6 +9,15 @@ def confirm_recently_authenticated
     prompt_for_current_password
   end
 
+  def confirm_recently_authenticated_2fa
```

Comment:
What if the condition checking the feature flag lived in `confirm_recently_authenticated`, then we wouldn't need all the extra code changes for controllers using this concern?

Reply:
Thanks for the suggestion, that's a great idea. Changed in 5223be8